python/ml4ir/applications/ranking/model/losses/listwise_losses.py

```diff
@@ -33,18 +32,7 @@ def _loss_fn(y_true, y_pred):
             y_pred : [batch_size, num_classes, 1]
             mask : [batch_size, num_classes, 1]
             """
-            batch_size = tf.cast(tf.shape(y_true)[0], tf.float32)
-
-            # Mask the padded records
-            y_true = tf.gather_nd(y_true, tf.where(tf.equal(mask, tf.constant(1.0))))
-            y_pred = tf.gather_nd(y_pred, tf.where(tf.equal(mask, tf.constant(1.0))))
-
-            # Reshape the tensors so that we sum the losses from each record
-            y_true = tf.expand_dims(tf.squeeze(y_true), axis=-1)
-            y_pred = tf.expand_dims(tf.squeeze(y_pred), axis=-1)
-
-            # Scale the sum of losses down by number of queries in the batch
-            return tf.math.divide(bce(y_true, y_pred), batch_size)
+            return cce(y_true, tf.math.multiply(y_pred, mask))
 
         return _loss_fn
 
```

Comment:
Now get_final_activation_op is part of another class. There is no breaking changes or tests about it?

Reply:
I am inheriting that class, right?

Reply:
Both the losses use the same activation. They just use a different method to compute the final loss.

Reply:
Oh I missed that. Do we want to have a common parent class for it? Or are we fine keeping it that way?

Reply:
I think it should be fine because Rank One ListNet loss is a variation of the softmax cross entropy loss, that's all.